streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -728,33 +745,57 @@ int maybeCommitActiveTasksPerUserRequested() {
         if (rebalanceInProgress) {
             return -1;
         } else {
-            final Map<TaskId, Map<TopicPartition, OffsetAndMetadata>> consumedOffsetsAndMetadataPerTask = new HashMap<>();
-            for (final Task task : activeTaskIterable()) {
-                if (task.commitRequested() && task.commitNeeded()) {
-                    task.prepareCommit();
-                    final Map<TopicPartition, OffsetAndMetadata> offsetAndMetadata = task.committableOffsetsAndMetadata();
-                    if (!offsetAndMetadata.isEmpty()) {
-                        consumedOffsetsAndMetadataPerTask.put(task.id(), offsetAndMetadata);
+            if (eosBetaEnabled && !eosUpgradeModeEnabled) {
+                for (final Task task : activeTaskIterable()) {
+                    if (task.commitRequested() && task.commitNeeded()) {
+                        return commitAll();
```

Comment:
For eos-beta, if we find a single task that we need to commit, we fall back to commit all tasks.

Reply:
I thought we agreed that we would always commitAll upon user requested in any task, in 2.6 and beyond, no matter of the eos config? Personally I think such if/else branching is not really worth the complexity itself..

Reply:
My understanding was that we only do this for eos-beta? \cc @abbccdda 

Don't really have a strong opinion what might be better.

Reply:
If you guys are not feeling strong, then let's just do this :)

Reply:
Ack